source/main/index.js

```diff
@@ -245,6 +247,40 @@ const onAppReady = async () => {
   app.on('before-quit', async (event) => {
     logger.info('app received <before-quit> event. Safe exiting Daedalus now.');
     event.preventDefault(); // prevent Daedalus from quitting immediately
+
+    if (isSelfnode) {
+      if (keepLocalClusterRunning) {
+        logger.info(
+          'ipcMain: Keeping the local cluster running while exiting Daedalus',
+          {
+            keepLocalClusterRunning,
+          }
+        );
+        return safeExitWithCode(0);
+      }
+
+      const exitSelfnodeDialogOptions = {
+        buttons: ['Yes', 'No'],
+        type: 'warning',
+        title: 'Daedalus is about to close',
+        message: 'Do you want to keep the local cluster running?',
+        defaultId: 0,
+        cancelId: 1,
+        noLink: true,
+      };
+      const { response } = await dialog.showMessageBox(
```

Comment:
Please enclose it with `try` and `catch` 'cos await is being used here.

Reply:
Thanks for the suggestion @yakovkaravelov - I have decided not to implement try/catch here since we do not use this patter elsewhere where we use Electron dialog.